packages/flutter/lib/src/material/checkbox.dart

```diff
@@ -554,12 +569,9 @@ class _RenderCheckbox extends RenderToggleable {
       ..strokeWidth = _kStrokeWidth;
   }
 
-  void _drawBorder(Canvas canvas, RRect outer, double t, Paint paint) {
+  void _drawBorder(Canvas canvas, Rect outer, double t, Paint paint) {
     assert(t >= 0.0 && t <= 0.5);
-    final double size = outer.width;
-    // As t goes from 0.0 to 1.0, gradually fill the outer RRect.
-    final RRect inner = outer.deflate(math.min(size / 2.0, _kStrokeWidth + size * t));
-    canvas.drawDRRect(outer, inner, paint);
+    shape.paint(canvas, outer);
```

Comment:
With using this method, I'm facing this issue:
If shape is passed null, then a checkbox with black border, width = 2 and borderRadius = 1 is rendered.
If shape is defined as 
```
shape: RoundedRectangleBorder(
   borderRadius: BorderRadius.circular(5),
),
```
Then a checkbox with transparent border and borderRadius = 5 is rendered (This should be a checkbox with border color, width = 2 and borderRadius = 5)

I want to fallback to default side if it isn't provided
How should I proceed?